lib/core/extension/proposal_details_model_extension.dart

```diff
@@ -39,27 +43,42 @@ extension ProposalDetailsModelExtension on ProposalDetailsModel {
     return null;
   }
 
-  // TODO(Saif): adjust this function
   String? tokenValue(int index, bool isOneCycleRewardsShown) {
-    String input;
```

Comment:
the model should store all values as numbers

if the values are strings, that's ok, just use accessors to access the number values

there shouldn't be any regexps anywhere here... ??!